kyaml/kio/byteio_reader.go

```diff
@@ -37,6 +38,9 @@ type ByteReadWriter struct {
 	// the Resources, otherwise they will be cleared.
 	KeepReaderAnnotations bool
 
+	// AddSeqIndentAnnotation if true adds kioutil.SeqIndentAnnotation to each resource
+	AddSeqIndentAnnotation bool
```

Comment:
There's also an awkward conflict between this option (however named) and `OmitReaderAnnotations`.  If we can do it cheaply enough, wdyt about treating it like any other reader annotation, and adding a "preserve indent style" option on the writer to make use of it where available? 

Related to this, I see that the writer already has a `Style` option, which I'm guessing might also interact with this new feature. E.g. if the `Style` is set to the json-like `FlowStyle` where sequence indent style doesn't apply, I'm guessing `Style` wins. We should document/test this.

Reply:
> There's also an awkward conflict between this option (however named) and `OmitReaderAnnotations`. If we can do it cheaply enough, wdyt about treating it like any other reader annotation, and adding a "preserve indent style" option on the writer to make use of it where available?
> 

It is not cheap to identify the indentation in byteio_reader. Hence made it opt-in. Except for that, this annotation is identical to `index` annotation and follows the same life cycle. 

> Related to this, I see that the writer already has a `Style` option, which I'm guessing might also interact with this new feature. E.g. if the `Style` is set to the json-like `FlowStyle` where sequence indent style doesn't apply, I'm guessing `Style` wins. We should document/test this.

That is a good point. Few options clash here.

Both `OmitReaderAnnotations` and `Style` win over `PreserveSequenceIndentStyle`. I will document and test this behavior.

Reply:
> Would it make sense to name this for the effect rather than the mechanism? E.g. something like `PreserveSequenceIndentStyle`

I started off with this but @droot thinks otherwise. I was on the fence. But the more I think, I think it is better to go with the mechanism rather than annotation(@droot WDYT). In that case, should we error out if conflicting options are set ? For e.g. `OmitReaderAnnotations` and `PreserveSequenceIndentStyle` are set ?

Reply:
Yes, `PreserveSequenceIndentStyle` makes sense at the `ReaderWriter` level or `PackageReaderWriter` level because they do reading and writing so can encapsulate the preserving behavior. So +1 to that. 
The same name may not make sense at the `Reader` level because reader can't ensure that behavior, so mechanism fits better there.

Reply:
Re: `OmitReaderAnnotations`. If we were starting today, would have definitely made the preserveIdentStyle as the default behavior and treating it like ReaderAnnotation. But at this point, to reduce the blast radius (and the cost involved), making this additive makes sense. There is an alternative path here, where we introduce this as an opt-in now (keeping it consistent with ReaderAnnotation) and eventually make it the default behavior.

Reply:
ByteReader is also exposed. We may change the logic even at bytereader as well. So it is better to be consistent.